index.js

```diff
@@ -98,16 +101,24 @@ function pickManifest (packument, wanted, opts) {
     packument.versions[target]
   )
   if (!manifest) {
-    err = new Error(
-      `No matching version found for ${packument.name}@${wanted}${
-        opts.enjoyBy
-          ? ` with an Enjoy By date of ${
-            new Date(opts.enjoyBy).toLocaleString()
-          }. Maybe try a different date?`
-          : ''
-      }`
-    )
-    err.code = 'ETARGET'
+    // Check if target is forbidden
+    const isForbidden = target && policyRestrictions && packument.policyRestrictions.restrictedVersions[target]
+    const pckg = `${packument.name}@${wanted}${
+      opts.enjoyBy
+        ? ` with an Enjoy By date of ${
```

Comment:
Curious. What is this 'Enjoy By' date about?

Reply:
TIL! https://www.npmjs.com/package/pacote#optsenjoy-by

Reply:
Interesting. So what's the benefit of filtering out packages that are newer than a given date?  Avoiding packages that haven't been thoroughly tested or widely deployed?

Reply:
@djsauble Reproducing installs as they would have happened at some point in the past.